github.go

```diff
@@ -278,6 +279,56 @@ func (m *GithubClient) UpdateCommitStatus(commitRef, baseContext, statusContext,
 	return err
 }
 
+func (m *GithubClient) DeletePreviousComments(prNumber string) error {
+	pr, err := strconv.Atoi(prNumber)
+	if err != nil {
+		return fmt.Errorf("failed to convert pull request number to int: %s", err)
+	}
+
+	var getComments struct {
+		Viewer struct {
+			Login string
+		}
+		Repository struct {
+			PullRequest struct {
+				Id       string
+				Comments struct {
+					Edges []struct {
+						Node struct {
+							DatabaseId int64
+							Author     struct {
+								Login string
+							}
+						}
+					}
+				} `graphql:"comments(last:$commentsLast)"`
+			} `graphql:"pullRequest(number:$prNumber)"`
+		} `graphql:"repository(owner:$repositoryOwner,name:$repositoryName)"`
+	}
+
+	vars := map[string]interface{}{
+		"repositoryOwner": githubv4.String(m.Owner),
+		"repositoryName":  githubv4.String(m.Repository),
+		"prNumber":        githubv4.Int(pr),
+		"commentsLast":    githubv4.Int(100),
+	}
+
+	if err := m.V4.Query(context.TODO(), &getComments, vars); err != nil {
+		return err
+	}
+
+	for _, e := range getComments.Repository.PullRequest.Comments.Edges {
+		if e.Node.Author.Login == getComments.Viewer.Login {
+			_, err := m.V3.Issues.DeleteComment(context.TODO(), m.Owner, m.Repository, e.Node.DatabaseId)
```

Comment:
I tried to use the graphql mutation `deleteIssueComment`, but was receiving an unhelpful error message. The v3 api worked immediately.